java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -422,6 +423,9 @@ abstract class AbstractArcHost(
     private suspend fun performParticleStartup(particleContexts: Collection<ParticleContext>) {
         if (particleContexts.isEmpty()) return
 
+        println("performParticleStartup")
```

Comment:
If you use log.debug instead of println you won't have to worry about removing these  println later.

Reply:
The general problem is that each particle has associated state in ParticleContext which lives on ArcHostContext. When various ArcHost operations occur that must manipulate this state (start, stop), it needs to locate the right one. 

The instance-id effectively is the index of the Plan.Particle in partition.particles, so maybe you could pass the index of the Plan.Particle in that list down through setupParticlesAndHandles, which could pass it to lookupParticleContextOrCreate() so it can always have a 1:1 mapping of Partition.particles to ArcHostContext.particles

However I am a little iffy on the sort ordering implications, since the Plan object could store the Particles in any order in the recipe, and if the code-gen were to change the order, it would be incompatible with the serialization. 

That is, say FooParticle is instantiated 3 times, Foo1, Foo2, Foo3, but then the user recompiles with the recipe modified so the order is 'Foo3, Foo2, Foo1', but a serialized Arc for this recipe has them stored in a List in the original order. This will get de-serialized wrong, and we currently have no check for this, e.g. store a hash of the Plan.Partition with the serialized arc to detect changes to the recipe. 

However, this is a more general problem of modifying recipes breaking serialization, since they could do far more than just modify the sort order.

Perhaps we can live with this until we put in error checking to version the arc serialization.

Reply:
I want to remove them :-D it was just to  prove to myself the cardinality differences.